src/Evolution/Systems/ScalarWave/Equations.cpp

```diff
@@ -27,6 +30,95 @@ void ComputeDuDt<Dim>::apply(
     dt_phi->get(d) = -d_pi.get(d);
   }
 }
+/// \endcond
+
+template <size_t Dim>
+void ComputeNormalDotFluxes<Dim>::apply(
+    const gsl::not_null<Scalar<DataVector>*> pi_normal_dot_flux,
+    const gsl::not_null<tnsr::i<DataVector, Dim, Frame::Inertial>*>
+        phi_normal_dot_flux,
+    const gsl::not_null<Scalar<DataVector>*> psi_normal_dot_flux,
+    const Scalar<DataVector>& pi,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& phi,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>&
+        interface_unit_normal) noexcept {
+  // We assume that all values of psi_normal_dot_flux are the same. The reason
+  // is that std::fill is actually surprisingly/disappointingly slow.
+  if (psi_normal_dot_flux->get()[0] != 0.0) {
+    std::fill(psi_normal_dot_flux->get().begin(),
+              psi_normal_dot_flux->get().end(), 0.0);
+  }
+
+  get(*pi_normal_dot_flux) = get<0>(interface_unit_normal) * get<0>(phi);
+  for (size_t i = 1; i < Dim; ++i) {
+    get(*pi_normal_dot_flux) += interface_unit_normal.get(i) * phi.get(i);
+  }
+
+  for (size_t i = 0; i < Dim; ++i) {
+    phi_normal_dot_flux->get(i) = interface_unit_normal.get(i) * get(pi);
+  }
+}
+
+template <size_t Dim>
```

Comment:
I got quite confused by what is stored in the packaged_data, please explicitly give the equation for what is stored by each tensor in the Vars in terms of the comment for the UpwindFlux struct.  And then you can remove the internal comment.